### PR TITLE
Update test for ckpt+val_check_interval

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

While reviewing https://github.com/PyTorchLightning/pytorch-lightning/pull/7032, noticed that this test is a bit incorrect since with total train batches=12 and val_check_interval=0.33, actual batch interval = 4 which aligns with the train_loop perfectly. In the test, we need to check for the case where val_check_interval doesn't align train batches, so updated the test here.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Good catch !

File: tests/checkpointing/test_model_checkpoint.py [hunk 1]
```diff
@@ -237,35 +240,55 @@ def configure_optimizers(self):
...
+    def _make_assertions(epoch, ix, add=''):
+        global_ix = ix + per_epoch_call_count * epoch
+        score = scores[global_ix]
+        expected_score = getattr(model, f'{monitor}s')[global_ix].mean().item()
+        expected_filename = f'{monitor}={score:.4f}-epoch={epoch}{add}.ckpt'
+        assert math.isclose(score, expected_score, rel_tol=1e-4)
+
+        chk = pl_load(os.path.join(checkpoint.dirpath, expected_filename))
+        assert chk['epoch'] == epoch + 1
+        assert chk['global_step'] == per_epoch_steps * (global_ix + 1) + (left_over_steps * (epoch + (1 if add else 0)))
```
Comment (carmocca) [inline tests/checkpointing/test_model_checkpoint.py:264]: nit: just for sanity, can you use some variables for the right side of this assertion?

**Review by carmocca — APPROVED** [review]

File: tests/checkpointing/test_model_checkpoint.py [hunk 2]
```diff
@@ -236,35 +239,55 @@ def configure_optimizers(self):
...
```
Comment (rohitgr7) [inline tests/checkpointing/test_model_checkpoint.py:263]:
```suggestion
        epoch_num = epoch + (1 if add else 0)
        expected_global_step = per_epoch_steps * (global_ix + 1) + (left_over_steps * epoch_num)
        assert chk['global_step'] == expected_global_step
```

File: tests/checkpointing/test_model_checkpoint.py (hunk 1 above)
Comment (rohitgr7) [reply tests/checkpointing/test_model_checkpoint.py]: done.

**Review by awaelchli — APPROVED** [review]

File: tests/checkpointing/test_model_checkpoint.py (hunk 1 above)
Comment (Borda) [reply tests/checkpointing/test_model_checkpoint.py]: @rohitgr7 pls when done, mar it also as resolved :]

**Review by Borda — APPROVED** [review]
lgtm

**Pull request merged**